File: 0744_find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.c
char nextGreatestLetter(char* letters, int lettersSize, char target){
    int left=0;
    int right=lettersSize-1;
    if(target>=letters[lettersSize-1]){
        return letters[0];
    }else{
        while(left<=right){
            int mid=(left+right)/2;
            if(letters[mid]==target){
                int a=mid;
                while(letters[a]==target){
                    a++;
                }
                return letters[a];
            }else if(letters[mid]<target){
                left=mid+1;
            }else{
                right=mid-1;
            }
        }
        return letters[left];
    }

}

File: 0744_find-smallest-letter-greater-than-target/find-smallest-letter-greater-than-target.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int ans = 100;
        for(char c : letters){
            if (c > target){
                ans = Math.min(ans, c - target);
            }
        }
        char result;
        if(ans == 100){
            result = letters[0];
        }else{
            result = (char)(target + ans);
        }
        return result;
    }
}